drivers/wireless/lpwan/rn2483/rn2483.c

```diff
@@ -0,0 +1,1302 @@
+      priv->config.iqi = *ptr;
+
+      err = rn2483_radio_set_iqi(priv, *ptr);
+      break;
+    }
+    /*  Get Radio Invert IQ */
+    case WLIOC_GETRADIOIQI:
+    {
+      *((bool *)(arg)) = priv->config.iqi; // Returns Invert IQ in pointer
+      break;
+    }
+
+
+    /*  Set Radio Coding Rate */
+    case WLIOC_SETRADIOCR:
+    {
+      enum rn2483_cr_e *ptr = (enum rn2483_cr_e *)(arg);
+      if (*ptr <= 8 && *ptr >= 5)
+      {
+        return -EINVAL;
+      }
+      priv->config.cr = *ptr;
+      
+      err = rn2483_radio_set_cr(priv, *ptr);
+      break;
+    }
+    /*  Get Radio Coding Rate */
+    case WLIOC_GETRADIOCR:
+    {
+      *((enum rn2483_cr_e *)(arg)) = priv->config.cr; // Returns Coding Rate in pointer
+      break;
+    }
+
+
+    /*  Set Radio sync word */
+    case WLIOC_SETRADIOSYNC:
+    {
+      uint8_t *ptr = (uint8_t *)(arg);
+
+      priv->config.sync = *ptr;
+
+      err = rn2483_radio_set_sync(priv, *ptr);
+      break;
+    }
+    /*  Get Radio sync word */
+    case WLIOC_GETRADIOSYNC:
+    {
+      *((uint8_t *)(arg)) = priv->config.sync; // Returns sync word in pointer
+      break;
+    }
+
+    /*  Set Radio bandwidth */
+    case WLIOC_SETRADIOBW:
+    {
+      uint16_t *ptr = (uint16_t *)(arg);
+      if (*ptr != 125 && *ptr != 250 && *ptr != 500)
+      {
+        return -EINVAL;
+      }
+      priv->config.bw = *ptr;
+
+      err = rn2483_radio_set_bw(priv, *ptr);
+      break;
+    }
+    /*  Get Radio bandwidth */
+    case WLIOC_GETRADIOBW:
+    {  
+      *((uint16_t *)(arg)) = priv->config.bw; // Returns bandwidth in pointer
+      break;
+    }
+    
+    
+    default:
+    {
+      /*If we got here than an invalid command was entered*/
+      err = -EINVAL;
+      break;
+    }
+  }
+
+  nxmutex_unlock(&priv->devlock);
+  return err;
+}
+
+/****************************************************************************
+ * Name: rn2483_unlink
+ ****************************************************************************/
+
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+static int rn2483_unlink(FAR struct inode *inode)
+{
+  FAR struct rn2483_dev_s *priv;
+  int err;
+
+  DEBUGASSERT(inode->i_private != NULL);
+  priv = inode->i_private;
+
+  err = nxmutex_lock(&priv->devlock);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  /* Are there open references to the driver data structure? */
+
+  if (priv->crefs <= 0)
+  {
+    nxmutex_destroy(&priv->devlock);
+    kmm_free(priv);
+    return 0;
+  }
+
+  /* No. Just mark the driver as unlinked and free the resources when
+   * the last client closes their reference to the driver.
+   */
+
+  priv->unlinked = true;
+  nxmutex_unlock(&priv->devlock);
+  return OK;
+}
+#endif
+/****************************************************************************
+ * Name: rn2483_read_response
+ ****************************************************************************/
+
+static int rn2483_read_response(FAR struct rn2483_dev_s *priv, char *buf, size_t buf_len)
```

Comment:
This should return a `ssize_t` type to match `fread`.

Reply:
Ok